fix: studio swagger api-docs crashes with 500 error

sumair-arbisoft opened the pull request:
## Description

Attempting to view the openapi (swagger) api docs for the cms fails. The Swagger web UI loads successfully, but displays a message "Failed to load API definition.".


## Supporting information

https://github.com/openedx/edx-platform/issues/37084


## Testing instructions

CMS API docs should load properly and no error logged on the backend.


## Other information

<img width="1552" height="987" alt="Screenshot 2025-07-31 at 9 20 11 PM" src="https://github.com/user-attachments/assets/e7d75fd7-d914-4d5f-9d70-8e60a2d9ec99" />

sumair-arbisoft committed:
fix: studio api-docs crashes with 500 error

hunzlahmalik commented on openedx/core/djangoapps/content_libraries/rest_api/containers.py:
```diff
@@ -303,6 +302,8 @@ class LibraryContainerCollectionsView(GenericAPIView):
     """
     View to set collections for a container.
     """
+    serializer_class = serializers.DummySerializer
```
We are using this serializer for the input data:
```
serializer = serializers.ContentLibraryItemCollectionsUpdateSerializer(data=request.data)
```

Can we use this for the serializer_class?

hunzlahmalik replied:
Also can we have define custom swagger response?

hunzlahmalik commented on openedx/core/djangoapps/content_libraries/rest_api/containers.py:
```diff
@@ -336,6 +337,8 @@ class LibraryContainerPublishView(GenericAPIView):
     """
     View to publish a container, or revert to last published.
     """
+    serializer_class = serializers.DummySerializer
```
For no input and no response can we use https://drf-yasg.readthedocs.io/en/stable/custom_spec.html instead of DummySerializer.

sumair-arbisoft replied:
if we are ok with excluding this endpoint then this option worked.
swagger_schema = None

hunzlahmalik replied:
I'm taking about `swagger_auto_schema` Which is already in use in the current file.

sumair-arbisoft committed:
fix: studio api-docs crashes with 500 error

sumair-arbisoft committed:
fix: studio api-docs crashes with 500 error

sumair-arbisoft committed:
fix: studio api-docs crashes with 500 error

sumair-arbisoft committed:
fix: studio api-docs crashes with 500 error

docs: Update codeowners to use normal CC teams (#37098)

See https://github.com/openedx/axim-engineering/issues/1511

Referenced teams:

https://github.com/orgs/openedx/teams/committers-edx-platform-2u-infinity/members
https://github.com/orgs/openedx/teams/committers-edx-platform-2u-aperture/members
https://github.com/orgs/openedx/teams/committers-edx-platform-2u-tnl/members

feat: include the usage_key in API to get ancestor blocks (#37072)

- includes location along with the child ancestor block as it is
required in the frontend authoring app for unit scrolling purpose in the
breadcrumbs.

Required for https://github.com/openedx/frontend-app-authoring/issues/1924

chore: Remove notification app waffle flags (#37086)

Update who gets tagged now that Arbi-bom is shutting down. (#37097)

* build: Update places where Arbi-bom was getting tagged.

The arbi-bom team no longer exists and we need to make sure these still
land in a timely manner.  So tag the maintainers to review and merge
these PRs

* build: Drop unnecessary common_constraint overrides.

These are no longer necessary now that
https://github.com/openedx/edx-lint/pull/500 has landed.

fix: updated captcha api to use enterprise assessment (#37079)

* fix: updated captcha api to use enterprise assessment

feat: disallow adding children to containers imported from library (#37088)

Updates `childAddable` action in xblocks that have upstream link to stop users from updating imported sections and subsections.

fix: fixed missing course in bulk delete (#37103)

chore: cleaned up course level preferences (#37106)

* chore: cleaned up course level preferences

* fix: fixed import issue

* fix: fixed lint errors

fix: Sort the output of dump_settings (#37118)

The dump_settings management command is
used for debugging changes to Django settings.
When comparing settings between two branches or
modules side-by-side, it is very useful to have the keys
print out in a deterministic order.

Affects developers only. This has no end-user impact.

build(deps): bump supercharge/mongodb-github-action

Bumps [supercharge/mongodb-github-action](https://github.com/supercharge/mongodb-github-action) from 1.11.0 to 1.12.0.
- [Release notes](https://github.com/supercharge/mongodb-github-action/releases)
- [Changelog](https://github.com/supercharge/mongodb-github-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/supercharge/mongodb-github-action/compare/1.11.0...1.12.0)

---
updated-dependencies:
- dependency-name: supercharge/mongodb-github-action
  dependency-version: 1.12.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

refactor: move production defaults to common modules (#37045)

In the effort to simplify settings in edx-platform, as discussed in ADR 22 -
Settings Simplification, this PR brings some of the production defaults defined
in `lms/envs/production.py` and `cms/envs/production.py` up to
`openedx/envs/common.py` or `lms/envs/common.py` and `cms/envs/common.py` as
appropriate.

Bringing these defaults up from the `production.py` settings modules caused
changes in the rendered settings of the `test.py` modules, and so I have
settings to the `test.py` modules to bring the rendered settings back in line
with what is has been. I have not deeply looked at which settings are needed
for tests to pass or not, but just the differences between the rendered
settings between `master` and this branch.

ADR 22: https://github.com/openedx/edx-platform/blob/master/docs/decisions/0022-settings-simplification.rst
Fixes https://github.com/openedx/edx-platform/issues/36892.

fix: remove enrollment notification until functionality is live (#37116)

* fix: remove enrollment notification until functionality is live

* fix: updated config version

* fix: fixed test cases

fix: Update vertical block test to use dynamic course id

chore: Upgrade Python requirements

chore: geoip2: update maxmind geolite country database

refactor: convert function based view to class based

fix: remove redundant code

temp: added logs for debugging in thread delete (#37127)

feat!: update finder method to work with django 42 and 52 (#37131)

feat: Send an LTI launch event for LTI Launches

feat: reset_student_attempts_for_entrance_exam to DRF (#37069)

* feat: reset_student_attempts_for_entrance_exam to DRF

* fix: update imports sequence

---------

Co-authored-by: Awais Qureshi <awais.qureshi@arbisoft.com>

Revert "chore: Upgrade Python requirements" (#37126)

fix: course detail page error (#37132)

rest_framework.fields.BuiltinSignatureError: Field source for `InstructorInfoSerializer.title` maps to a built-in function type and is invalid. Define a property or method on the `str` instance that wraps the call to the built-in function.

Co-authored-by: Muhammad Faraz  Maqsood <faraz.maqsood@A006-01130.local>
